src/opencl_rawsha512_gpl_fmt_plug.c

```diff
@@ -58,6 +58,11 @@ static cl_mem pass_buffer;      //Plaintext buffer.
 static cl_mem idx_buffer;       //Sizes and offsets buffer.
 static cl_kernel prepare_kernel;
 
+//A list containing all OpenCL buffers created by the format.
+static cl_mem cl_buffer_list[CL_MAX_BUFFERS];
+static void *cl_map_list[CL_MAX_BUFFERS][2];
+static int cl_buffer_size, cl_map_size;
```

Comment:
Should we possibly provide such helpers that we'd only need to define one variable here (of a custom struct type) and the helpers would handle allocation and freeing of the buffer list dynamically? Perhaps it can be a linked list managed by the helpers.

Reply:
Something like that is boiling in my head too: A single define for all buffers that depend on GWS. Then a single call to allocate all of them and a single call to destroy them.

Perhaps even a single define for all buffers, where some depend on GWS and some don't? We need to keep scope in mind though.

What is the scope? I think it's "minimize (as in share) code that all formats need", but also - and this is important: "make things easy to read and follow".